index.bs

```diff
@@ -278,6 +278,35 @@ To <dfn abstract-op lt="set-user">set the `Sec-Fetch-User` header</dfn> for a [=
 </ol>
 </div>
 
+The `Sec-Purpose` HTTP Request Header {#sec-purpose-header}
```

Comment:
Given the existing implementation, I'll simply bemoan the fact that we're neither consistent with the rest of fetch metadata (`Sec-Fetch-*`) nor with Fetch's terminology (`initiator`). That's unfortunate, and perhaps not too late to change?

Reply:
I think it's not too bad because purpose is not exactly a fetch concept. But I really don't mind. @domenic?

Reply:
I originally defined it in HTML, but then it needs special handling so that it doesn't create a preflight etc. This seemed like the place where we define these `Sec-` headers so it felt natural. Alternatives?

Reply:
Oh, is this the spec that defines special handling for preflights?

Reply:
It's not really. This only defined the `Sec-Fetch-*` headers to date. And those could be merged into Fetch if we wanted to.

The main thing most `Sec-*` headers need to adhere to is that they are bound to the request rather late, post-SW.

Reply:
It looks like it is not; Fetch defines that. So it seems to me defining the header in Fetch would make the most sense, if it's tied to initiator. @annevk's thoughts are welcome.

As for exposing a general initiator header vs. what's specified here: I think what's specified here is better. Initiator is a spec-internal term which we could easily change, and/or might be going away in the future: https://github.com/whatwg/fetch/issues/1563 . And we should expose the minimum amount of information, not all possible initiators, so that we don't end up in the same situation as Sec-Fetch-Dest where apparently we're not sure what people are doing with the values we expose.

As for whether we should prefix with Sec-Fetch- vs. Sec-: I don't really understand the purpose of the Fetch- prefix, since all headers on the web platform are sent by fetch, so it feels like a legacy mistake to me that we wouldn't want to repeat to include that redundant prefix.

Reply:
Is my comment above not visible? Anyway, I concur.

Reply:
It's visible but I didn't read it as opinionated.
Anyway, all clarified now.